Add metab admin form to add/remove aliases.

hjarrell opened the pull request:
**What does this change do?** _Please be clear and concise._

Add a form which allows you to create a new alias or delete an existing
alias. Also changed the names table to extend the unique constraint to a withheld
boolean as well. This means that when you mark someone withheld you can still use
their name as an alias. Also update change user's withheld to mark all names as the same
updated withheld status.

**Why was this change made?** _Including an issue number is sufficient. Otherwise, briefly explain the benefit of the change._

Completes SEP-133


## Verification

_List the steps needed to make sure this thing works. Be precise._

1. Setup a local supplemental database by following the steps in the README.
1. Start a local metab_admin instance with the steps in the README
1. Fill out the alias form
1. Verify that the data in the database match (names withholding changes from true/false, names get added).

- Verify the thing does this:
    - Ensures that active aliases are unique
- Pictures:

![image](https://user-images.githubusercontent.com/3639154/67789941-3a5a5300-fa4b-11e9-834e-313a5d623d2f.png)

## Affirmations

All of these should have a check by them. Any exception requires an explanation.

* [x] I matched the style of the existing code.
* [x] I added and updated any relevant documentation (inline, `README`, `CHANGELOG`, and such).
* [x] I used Python's type hinting.
* [x] I ran the automated tests and ensured they **ALL** passed.
* [x] I ran the linter and ensured my changes have not introduced **ANY** warnings or errors.
* [x] I have made an effort to minimize code changes and have not included any cruft (preference files, *.pyc files, old comments, print-debugging, unused variables).
* [x] I have made an effort maintain a clear commit history (haven't merged other branches or rebased improperly).
* [x] I have written the code myself or have given credit where credit is due.

hjarrell committed:
Add metab admin form for adding and removing aliases.

Add a form which allows you to create a new alias or delete an existing
alias. Also changed the names table to extend the unique constraint to a witheld
boolean as well. This means that when you mark someone withheld you can still use
their name as an alias. Also update change user's withheld to mark all names as the same
updated withheld status.

hjarrell committed:
Reset deleted gitignore file

ufl-taeber commented:
Be sure to add *After* and *Before* screenshots when the UI changes. In this case it'd just be *After*.

ufl-taeber commented:
After an error occurs, the checkbox still indicates the person is withheld when they are not. I'd prefer if it reverted to the previous state.

![image](https://user-images.githubusercontent.com/7585041/67706709-a1b2cd00-f98f-11e9-91ba-e03abb786a71.png)

hjarrell committed:
Revert checkbox state on error for withheld people form.

Revert the checkbox back to the checked state before it was clicked
when there is an error updating the withheld status for a person. This
ensures that the UI matches the actual state of the database/your actions.

hjarrell commented:
@ufl-taeber I fixed it so that the checkbox will revert back to the state before the error message.

hjarrell committed:
Fix sql query in metab import for names withheld.

Update a sql query on metab import which selects people and names
that are not withheld. The withheld column was ambiguous.
